* Introduction

Originally I was trying to collect input data for our C1 paper.

We had a final filter set of 433 cells. Since there was new data, 
I then committed my original code [[orgit-rev:/ssh:pongo:/woldlab/castor/home/diane/proj/C1_mouse_limb_combined/::0c9d57e][/ssh:pongo:proj/C1_mouse_limb_combined/ (magit-rev 0c9d57e)]]

* Create publication set at the DCC

[[file:find_experiment.sql::\pset%20format%20unaligned][find_experment.sql]] generates [[file:publication_files.tsv::experiment_accession%20experiment_description%20file%20filename%20file_type%20term_name%20age%20age_units%20library%20biosample%20library_created][publication_files.tsv]] which was 
used by [[file:create_publication_set.py][file:create_publication_set.py]] plus
associate all of the fastqs from our 433 set with the encode publication
https://www.encodeproject.org/publications/e0d01543-9965-4edb-933c-778a40575cd9/

* Request <2018-03-21 Wed>

Today I receved a request to compute our coverage "metric" for all of the new data.
The coverage metric was computing the slope of the middle of the coverage plot
[[file:/ssh:pongo:/woldlab/castor/home/diane/proj/long-rna-seq-condor/woldrnaseq/plot_coverage.py::def%20make_median_normalized_summary(experiments,%20coverage):][Coverage slope plot]]

I added the "new" list of cells to [[file:to_include.py::def%20generate_to_include_as_of_run17():][generate_to_include_as_of_run17]]

The information I received from Peng suggested I needed to compute the
slope for all the data, so we could find any outliers

And that suggests making a histogram of the slopes.

I just talked to Barbara and she wants the plots for each of the runs.

I unfortunately don't have a map between run & library.

** Implementation

First step was to generate combined FPKM and coverate quantification files.
(First attempt)
Run [[file:combine_coverage.py][combine_coverage.py]] to generate [[file:C1_mouse_limb_coverage_asof_run17.tsv][C1_mouse_limb_coverage_asof_run17.tsv]]

Compute the slope using [[file:compute_slope.py][compute_slope.py]]

* Request <2018-03-23 Fri>

Brian had a spread sheet [[file:Slope_match_PacBio_cells_Diane_March23_2018.xlsx][Slope_match_PacBio_cells_Diane_March23_2018.xlsx]] and he wanted
me to merge the two columns.

Which I did with [[file:slope_match.py][slope_match.py]].

However he also wanted me to generate the genes-detected values for
the C1 mouse limb data which took me a while, because I needed to ask
Sean which versions of the source genes-detected files to use.

